dev/README.md

```diff
@@ -15,56 +15,98 @@ This container is a PostgreSQL DB. DB data is kept in a volume, persistent acros
-    This is the name of the directory used as a bind mount in the compose file.
+To start the `worker-manager`, [you need to register a worker](#registering-a-worker). Afterwards, run:
+```sh
+docker compose --profile worker up --build
+```
+
+## Registering a Worker
+
+- [Get wireguard configuration files](#get-wireguard-configuration-files)
+- [Create a test worker](#create-a-test-worker)
 
--  Move the Mullvad configuration files into the `data` directory. Configuration files should be named in the format `<country_code>.conf` where
+
+**NOTE:** Unless otherwise stated, all files and commands are with respective to the `dev` directory.
+
+
+### Get wireguard configuration files
+
+-  Move the [Mullvad](https://mullvad.net/) configuration files into the `data` directory.
+    Configuration files should be named in the format `<country_code>.conf` where
     `<country_code>` is the ISO 3166-1 alpha-2 code of the country.
-    On first time start up, the `worker_manager` will select a random file to start up the `wireguard` container.
+    On start, the `worker-manager` will select a random file to start up the `wireguard` container.
     On receiving a task from the backend scheduler, it will search for a configuration file belonging to that task and reconfigure the wireguard container.
     If the configuration file does not exist, it will skip the test.
 
-### create a test worker
+## create a test worker
 
 - Generate a private key:
     ```sh
     openssl genrsa -out id_rsa 2048
     ```
     The key name `id_rsa` is used as a bind mount in the compose file.
 
-- Assuming the backend service is up, create a worker in the backend:
+- Assuming the backend service is up (`docker compose up backend`), create a worker and assign them a list of countries to test for.
+    If an existing worker is responsible for those countries, they are assigned to the new worker. In this example, we create a worker
+    named `test`.
     ```sh
-    docker exec -i mirrors-qa-backend mirrors-qa-backend create-worker --countries=us,fr,ca test_worker < ./id_rsa
-    export WORKER_ID=test_worker
+    docker exec -i mirrors-qa-backend mirrors-qa-backend create-worker --countries=us,fr,ca test < ./id_rsa
     ```
+- Set the name of the worker to the `WORKER_ID` variable in the `.env` file.
 
-### build the task image
-- Build the task worker image.
-    ```sh
-    cd ../worker/task
-    docker build -t mirrors-qa-task-worker .
-    export TASK_WORKER_IMAGE=mirrors-qa-task-worker
-    ```
+- Start the services with the worker enabled using `docker compose up --profile worker up --build`
+
+## Environment variables
+
+**NOTE:** All environment variables ending with a `_DURATION` suffix accept values that would be valid for [humanfriendly](https://humanfriendly.readthedocs.io/en/latest/api.html#humanfriendly.parse_timespan)
+
+### backend
+The `backend` code houses the `scheduler` and the `RESTful API`. The following environment variables are shared by both services:
```

Comment:
Leave a space after titles and before list ; otherwise breaks rendering on non-GH markdown